src/formats.ts

```diff
@@ -62,6 +69,21 @@ export const fullFormats: DefinedFormats = {
   "json-pointer-uri-fragment": /^#(?:\/(?:[a-z0-9_\-.!$&'()*+,;:=@]|%[0-9a-f]{2}|~0|~1)*)*$/i,
   // relative JSON-pointer: http://tools.ietf.org/html/draft-luff-relative-json-pointer-00
   "relative-json-pointer": /^(?:0|[1-9][0-9]*)(?:#|(?:\/(?:[^~/]|~0|~1)*)*)$/,
+  // the following formats are used by the openapi specification: https://spec.openapis.org/oas/v3.0.0#data-types
+  // byte: https://github.com/miguelmota/is-base64
+  byte: /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/gm,
+  // signed 32 bit integer
+  int32: validateInteger(32),
+  // signed 64 bit integer
+  int64: validateInteger(64),
+  // C-type float
+  float: validateNumber(128),
+  // C-type double
+  double: validateNumber(1024),
```

Comment:
I think this could be defined as `{type: "number", validate: isFinite}`, although `type: "number"` would do it already, so maybe just `true`

Reply:
From a javascript perspective yes, but if you validate them before export then a receiver using C will see a difference between floats and doubles. This was the only test I found to check the difference between floats and doubles.
We can set it to true, but it might come back to haunt us ;-)